### ISSUE TITLE
Numeric limits does not support __uint128 on gcc

### ISSUE THREAD

**Issue by sgerbino** [koinos/koinos-chain#6]
[This assert](https://github.com/open-orchard/koinos/blob/master/libraries/chain/include/koinos/chain/contract_table_objects.hpp#L168) is triggering under GCC.

Apparently GCC defines the `max()` to be zero for `__uint128`.

We have some options:

- (a) Use `__uint128`, attempt to fix via override
- (b) Use FC `uint128`
- (c) Define our own `uint128` type
- (d) Delete `uint128` entirely, don't use it at all

Basically I think we don't need a 128-bit *numeric* type at all outside the VM.  (Inside the VM it is probably pretty useful.)

We may want a 128-bit fixed-length data type for keys, but I'd be in favor of making that its own type which is only used for key comparisons.
